specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/SessionPools.json

```diff
@@ -634,16 +642,56 @@
                 "name": "SessionPoolProvisioningState",
                 "modelAsString": true
               }
+            },
+            "managedIdentitySettings": {
+              "description": "Optional settings for a Managed Identity that is assigned to the Session pool.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ManagedIdentitySetting"
+              },
+              "x-ms-identifiers": [
+                "identity"
+              ]
             }
           },
           "x-ms-client-flatten": true
         }
       }
     },
+    "ManagedIdentitySetting": {
+      "description": "Optional settings for a Managed Identity that is assigned to the Session pool.",
+      "type": "object",
+      "required": [
+        "identity"
+      ],
+      "properties": {
+        "identity": {
+          "description": "The resource ID of a user-assigned managed identity that is assigned to the Session Pool, or 'system' for system-assigned identity.",
+          "type": "string"
+        },
+        "lifecycle": {
+          "description": "Use to select the lifecycle stages of a Session Pool during which the Managed Identity should be available.",
+          "enum": [
+            "None",
+            "Main"
```

Comment:
In current Session scenario, suppose `Main` indicates the whole lifecycle, right? Am thinking whether we should use `All` or `Main` here, considering the different iteration story of [`Main`, `None`] -> [`Main`, `None`, `All`] vs [`All`, `None`] -> [`All`, `None`, `Main`]. Would like you hear more of your thoughts

Reply:
From PM "for sessions, if we have pullImage and main, then if someone selects main, it's not clear if that includes image pull as well 
i think none and main sound good for sessions"